pkg/lib/registry/registry.go

```diff
@@ -523,3 +445,75 @@ func isDeprecated(ctx context.Context, q *sqlite.SQLQuerier, bundle registry.Bun
 	}
 	return false, nil
 }
+
+/* opm index add drops bundles that replace a channel head, and since channel head selection heuristics
+* choose the bundle with the greatest semver as the channel head, any bundle that replaces such a bundle
+* will be dropped from the graph following an add.
+* eg: 1.0.1 <- 1.0.1-new
+*
+* 1.0.1-new replaces 1.0.1 but will not be chosen as the channel head because of its non-empty pre-release version.
+* expectedGraphBundles gives a set of bundles (old bundles from the graphLoader and the newly added set of bundles from
+* imagesToAdd) that must be present following an add to ensure no bundle is dropped.
+*
+* Overwritten bundles will only be verified on the channels of the newly added version.
+* Any inherited channels due to addition of a new bundle on its tail bundles may not be verified
+* eg:  [1.0.2 (alpha, stable)] <- 1.0.1 (alpha)
```

Comment:
The direction of the arrow in this example denotes "is replaced by", which is the opposite of the previous example's "replaces". I think we should keep the meaning uniform across examples.